backend/connector-integration/src/utils.rs

```diff
@@ -44,5 +89,2567 @@ pub fn missing_field_err(
     })
 }
 
-pub mod xml_utils;
-pub use xml_utils::preprocess_xml_response_bytes;
+type Error = error_stack::Report<errors::ConnectorError>;
+
+pub trait AccessTokenRequestInfo {
+    fn get_request_id(&self) -> Result<Secret<String>, Error>;
+}
+
+impl AccessTokenRequestInfo for types::RefreshTokenRouterData {
+    fn get_request_id(&self) -> Result<Secret<String>, Error> {
+        self.request
+            .id
+            .clone()
+            .ok_or_else(missing_field_err("request.id"))
+    }
+}
```

Comment:
```suggestion
pub trait AccessTokenRequestInfo {
    fn get_request_id(&self) -> Result<Secret<String>, Error>;
}

impl AccessTokenRequestInfo for types::RefreshTokenRouterData {
    fn get_request_id(&self) -> Result<Secret<String>, Error> {
        self.request
            .id
            .clone()
            .ok_or_else(missing_field_err("request.id"))
    }
}
```

Is this trait is required? else this can just a impl based function RefreshTokenRouterData